### PR TITLE
docs(swizzling): add useBlogPost to guide

### PR DESCRIPTION
**Pull request by wood3n**
In my experience, we need `useBlogPost` to guide how to add comment to `BlogPostItemWrapper`, this hook can determine that it is currently in blog post list or blog page.

## Test Plan
don't need test.

### CODE REVIEW

File: website/docs/swizzling.md [hunk 1]
```diff
@@ -207,12 +207,16 @@ Wrapping a theme is a great way to **add extra components around existing one**
 import React from 'react';
 import BlogPostItem from '@theme-original/BlogPostItem';
 import MyCustomCommentSystem from '@site/src/MyCustomCommentSystem';
+import { useBlogPost } from '@docusaurus/theme-common/internal':
```
Comment (Josh-Cena) [inline website/docs/swizzling.md:210]: We don't document internal hooks in docs, so waiting for @slorber to figure out a better way to handle this.

Comment (wood3n) [reply website/docs/swizzling.md:210]: ok, I think maybe it is the most simple way.

**Comment by slorber** [comment]
That's something we'll likely want to add in the future, but it's a bit odd to mention this internal API that we don't even document properly. Let's first focus on moving these internal apis to public apis and documenting them.

**Comment by wood3n** [comment]
> That's something we'll likely want to add in the future, but it's a bit odd to mention this internal API that we don't even document properly. Let's first focus on moving these internal apis to public apis and documenting them.

I think this hook is very useful to add a custom comment component under each blog post, and I can't find any other hooks that do the same.

**Pull request closed without merging**